Description: Ensure that the JNI interface is loaded
Author: Kumar Appaiah <akumar@debian.org>
Forwarded: not-needed
--- a/src/magick/Magick.java
+++ b/src/magick/Magick.java
@@ -15,28 +15,7 @@
 public class Magick {
 
     static {
-        String clprop = System.getProperty("jmagick.systemclassloader");
-        if (clprop == null || clprop.equalsIgnoreCase("yes")) {
-            try {
-                ClassLoader.getSystemClassLoader()
-                    .loadClass("magick.MagickLoader").newInstance();
-            }
-            catch(ClassNotFoundException e) {
-                throw new RuntimeException("Can't load MagickLoader " +
-                                           "(class not found)");
-            }
-            catch(IllegalAccessException e) {
-                throw new RuntimeException("Access to SystemClassLoader "+
-                                           "denied (IllegalAccessException)");
-            }
-            catch(InstantiationException e) {
-                throw new RuntimeException("Can't instantiate MagicLoader " +
-                                           "(InstantiationException)");
-            }
-        }
-        else {
-            System.loadLibrary("JMagick");
-        }
+        System.loadLibrary("JMagick");
         init();
     }
 
